src/main/scala/org/broadinstitute/dsde/workbench/sam/Boot.scala

```diff
@@ -223,7 +225,11 @@ object Boot extends IOApp with LazyLogging {
       googleKms: GoogleKmsService[IO],
       adminConfig: AdminConfig)(implicit actorSystem: ActorSystem): GoogleExtensions = {
     val workspaceMetricBaseName = "google"
-    val googleDirDaos = (config.googleServicesConfig.adminSdkServiceAccounts match {
+    val googleDirDaos = config.googleServicesConfig.adminSdkServiceAccountPaths.map(
+      _.map(path =>
+        Json(Files.readAllLines(Paths.get(path)).asScala.mkString,
+          Option(config.googleServicesConfig.subEmail))
+      )).getOrElse(config.googleServicesConfig.adminSdkServiceAccounts match {
```

Comment:
There's a lot going on here.  I'm inclined to take everything from 228-243 and extract it into its own method, and maybe create some intermediate variables.  Doesn't _need_ to happen, I'll 👍 the PR anyway, but I think it would be nice.

Reply:
Cleaned things up and now its much more readable!